mailet/api/src/main/java/org/apache/mailet/MailetContext.java

```diff
@@ -338,6 +339,21 @@ void sendMail(MailAddress sender, Collection<MailAddress> recipients, MimeMessag
     void sendMail(Mail mail)
             throws MessagingException;
 
+    /**
+     * Sends an outgoing message to the top of this mailet container's root queue.
+     * This is the equivalent of opening an SMTP session to localhost.
+     * The Mail object provides all envelope and content information
+     * 
+     * The given delay and unit are used to calculate the time when
+     * the Mail will be available for deQueue
+     *
+     * @param mail - the message that is to sent
+     * @param delay - the delay value for deQueue
+     * @param unit - the delay unit for deQueue
+     * @throws MessagingException if an error occurs accessing or sending the message
+     */
+    void sendMail(Mail mail, long delay, TimeUnit unit)
```

Comment:
You may replare _delay_ & _unit_ by a **java.time.Duration**.

This seems to be a more global task related to _MailQueue_, can you create a task for this?

Reply:
> You may replare delay & unit by a java.time.Duration.

I'd like a builder like fluent stuff here to avoid method duplication. Not for today ;-)

> This seems to be a more global task related to MailQueue, can you create a task for this?

Interesting! 

Remember that a mailet container should be able to be usable in a non James context!

That is why MailetContext replicate a lot of James API, including the MQ (which of course exposes delays). It is more of a wiring stuff, and that feature is exposed at the MQ level.

That being said, I still believe that PR adds value to the product.